### ISSUE TITLE
Support disabling consistent read on per @Query

### ISSUE THREAD

**Issue by dragneelfps** [boostchicken/spring-data-dynamodb#9]
## Expected Behavior
We should be able to disable consistent read behaviour on per query basis, instead of having to use a custom repository. 
This feature is helpful for queries on GSI as they dont support consistent reads.

## Actual Behavior
Either disable it for all using dynamodb mapper config or use custom repositories.

## Specifications
  
  - Spring Data DynamoDB Version: 5.1.0
  - Spring Data Version: 2.2.3.Release
  - AWS SDK Version: 1.11.443
  - Java Version: openjdk version "11.0.5" 2019-10-15
  - Platform Details: Macbook Pro mid-2018

**Comment by dragneelfps** [comment]
https://github.com/derjust/spring-data-dynamodb/issues/276

**Comment by boostchicken** [comment]
Live on 5.2.2-SNAPSHOT on ossrh

```@Query(consistentReads = QueryConstants.ConsistentReadMode.CONSISTENT)
	Optional<User> findById(String id);
```

Default does whatever the mapper has configured and is the default.  Other keys are self explanatory.

Leaving issue open for comments / suggestions

**Comment by boostchicken** [comment]
https://github.com/boostchicken/spring-data-dynamodb/commit/164de9846b5a96add316c6a72478375f8512a25b

**Comment by boostchicken** [comment]
@dragneelfps you get a chance to look at it?

**Comment by dragneelfps** [comment]
@boostchicken do you have any plans on creating a PR for the parent repository of `derkjust`.

Ill look into the changes and report back as soon as possible.

**Comment by dragneelfps** [comment]
@boostchicken instead of having an enum, can't we have a `boolean` as the original `private Boolean consistentRead;` in `QueryRequest` is a `boolean`?? I think this will be more consistent behavior.

**Comment by boostchicken** [comment]
The problem with this is you have to put it on every annotation.  Annotations can't have null values, they just have a default. By putting it to an enum we can have it be a default value which means change nothing. If we were to have an Boolean it would require ever you to adjust every query annotation and break the API significantly.

**Comment by dragneelfps** [comment]
Valid point.
But just a thought shower, how about 
```
boolean consistentReads() default true;
boolean override() default false;
```

**Comment by boostchicken** [comment]
I think its a little more verbose and confusing, only one annotation argument should be required.  Also, a lot of people may never want to default to true or false.  The default setting makes a lot of sense in my book

**Comment by boostchicken** [comment]
Released in 5.2.2
